### PR TITLE
upgrade minSDk , complieSDK and targetSDK

### PR DESCRIPTION
**Pull request by CharmisG**
Upgraded Plugin : 

-  minSDk to 24
- complieSDK and targetSDK to 30
- Resolved user issue :  https://github.com/square/in-app-payments-react-native-plugin/issues/179

### CODE REVIEW

File: react-native-in-app-payments-quickstart/index.js [hunk 1]
```diff
@@ -14,8 +14,9 @@
  limitations under the License.
 */
 
-import { AppRegistry } from 'react-native';
+import { AppRegistry, LogBox } from 'react-native';
 import App from './App';
 import { name as appName } from './app.json';
 
+LogBox.ignoreAllLogs();
```
Comment (hukid) [inline react-native-in-app-payments-quickstart/index.js:21]: Why do we do this?

Comment (CharmisG) [reply react-native-in-app-payments-quickstart/index.js:21]: One of our customer raised issue and I resolved this,
https://github.com/square/in-app-payments-react-native-plugin/issues/179

Comment (hukid) [reply react-native-in-app-payments-quickstart/index.js:21]: Can you add comment to this line explaining why this fix that issue? Is it a workaround to hide the error message or it's a real fix?

Comment (CharmisG) [reply react-native-in-app-payments-quickstart/index.js:21]: It's the real solution.

**Review by hukid — APPROVED** [review]

**Review by pedrocmello — APPROVED** [review]

**Comment by pedrocmello** [comment]
can we get this PR merged?

**Pull request merged**